src/ParallelRun.jl

```diff
@@ -296,8 +299,388 @@ function load_images(rcfs, stagedir)
+    reserve_thread[] && thread_fun(reserve_thread)
+
+    Log.info("Finding neighbors")
+    tic()
+    neighbor_map = Infer.find_neighbors(target_sources, catalog, images)
+    Log.info("Neighbors found in $(toq()) seconds")
+
+    reserve_thread[] && thread_fun(reserve_thread)
+
+    # Partition the sources
+    n_sources = length(target_sources)
+    Log.info("Optimizing $(n_sources) sources")
+    if cyclades_partition
+        # Convert neighbormap to cyclades map (map from source_id -> [neighbor source_ids])
+        cyclades_neighbor_map = Dict{Int64, Vector{Int64}}()
+        for (index, neighbors) in enumerate(neighbor_map)
+            source_id = target_sources[index]
+            cyclades_neighbor_map[source_id] = neighbors
+        end
+        thread_sources_assignment = partition_cyclades(nprocthreads, target_sources, cyclades_neighbor_map)
+    else
+        thread_sources_assignment = partition_equally(nprocthreads, n_sources)
+    end
+
+    Log.info("Done assigning sources to threads for processing")
+
+    # Pre allocate elbo args variational params
+    target_source_variational_params = Dict{Int64, Array{Float64}}()
+    for target_source in target_sources
+        cat = catalog[target_source]
+        target_source_variational_params[target_source] = init_source(cat)
+    end
+    
+    # Pre-allocate dictionary of elboargs, call it model.
+    model = Array{ElboArgs}(n_sources)
+    function initialize_elboargs_sources(sources)
+        nputs(dt_nodeid, "Thread $(Threads.threadid()) allocating mem for $(length(sources)) sources")
+        for cur_source_index in sources
+            entry_id = target_sources[cur_source_index]
+            entry = catalog[target_sources[cur_source_index]]
+            neighbor_ids = neighbor_map[cur_source_index]
+            neighbors = catalog[neighbor_map[cur_source_index]]
+
+            # TODO max: refactor this portion? It's reused in infer_source.
+            nputs(dt_nodeid, "Thread $(Threads.threadid()) allocating mem for source $(target_sources[cur_source_index]): objid=$(entry.objid)")
+            cat_local = vcat(entry, neighbors)
+            ids_local = vcat(entry_id, neighbor_ids)
+            
+            #vp = Vector{Float64}[init_source(ce) for ce in cat_local]
+            vp = Vector{Float64}[haskey(target_source_variational_params, x) ? target_source_variational_params[x] : init_source(catalog[x]) for x in ids_local]
+            patches, tile_source_map = Infer.get_tile_source_map(images, cat_local)
+            ea = ElboArgs(images, vp, tile_source_map, patches, [1])
+            Infer.fit_object_psfs!(ea, ea.active_sources)
+            Infer.load_active_pixels!(ea)
+            @assert length(ea.active_pixels) > 0
+            model[cur_source_index] = ea
+        end
+    end
+
+    # Initialize elboargs in parallel
+    tic()
+    thread_initialize_sources_assignment = partition_equally(nprocthreads, n_sources)
+    
+    Threads.@threads for i=1:nprocthreads
```

Comment:
Do you how much of a speedup you get here for using 2,4, and 8 threads, rather than 1 thread? One issue we've had is that Julia's GC is, for now, single-threaded, so the speedup often isn't as good as we'd like. The overhead is particularly bad for parts of the code that allocate and free a lot of memory.